spotify_dl/youtube.py

```diff
@@ -10,39 +10,54 @@
-        for v in search_response['items']:
-            if v['id']['kind'] == VIDEO:
-                log.debug("Adding Video id {}".format(v['id']['videoId']))
-                return YOUTUBE_VIDEO_URL + v['id']['videoId']
-    except HttpError as err: 
-        err_details = loads(err.content.decode('utf-8')).get('error').get('errors')
-        secho("Couldn't complete search due to following errors: ", fg='red')
-        for e in err_details:
-            error_reason = e.get('reason')
-            error_domain = e.get('domain')
-            error_message = e.get('message')
+    if scrape or not dev_key:
+        YOUTUBE_SEARCH_BASE = "https://www.youtube.com/results?search_query="
+        try:
+            response = html.fromstring(requests.get(YOUTUBE_SEARCH_BASE + search_term).content)
+            video = response.xpath("//a[contains(@class, 'yt-uix-tile-link')]/@href")
+            video_id = re.search("((\?v=)[a-zA-Z0-9_-]{4,15})", video[0]).group(0)[3:]
+            print(video_id)
+            return YOUTUBE_VIDEO_URL + video_id
+        except:
+            print("Could not perform scrape search")
+            return None
+    else:
+        YOUTUBE_DEV_KEY = dev_key
+        youtube = build(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION,
+                        developerKey=YOUTUBE_DEV_KEY,
+                        cache_discovery=False)
+        try:
+            search_response = youtube.search().list(q=search_term,
+                                                    part='id, snippet').execute()
+            for v in search_response['items']:
+                if v['id']['kind'] == VIDEO:
+                    log.debug("Adding Video id {}".format(v['id']['videoId']))
+                    return YOUTUBE_VIDEO_URL + v['id']['videoId']
+        except HttpError as err:
+            err_details = loads(err.content.decode('utf-8')).get('error').get('errors')
+            secho("Couldn't complete search due to following errors: ", fg='red')
+            for e in err_details:
+                error_reason = e.get('reason')
+                error_domain = e.get('domain')
+                error_message = e.get('message')
 
-            if error_reason == 'quotaExceeded' or error_reason == 'dailyLimitExceeded':
-                secho(f"\tYou're over daily allowed quota. Unfortunately, YouTube restricts API keys to a max of 10,000 requests per day which translates to a maximum of 100 searches.", fg='red')
-                secho(f"\tThe quota will be reset at midnight Pacific Time (PT)." ,fg='red')
-                secho(f"\tYou can request for Quota increase from https://console.developers.google.com/apis/api/youtube.googleapis.com/quotas.", fg='red')
-            else:
-                secho(f"\t Search failed due to {error_domain}:{error_reason}, message: {error_message}")
-        return None
+                if error_reason == 'quotaExceeded' or error_reason == 'dailyLimitExceeded':
```

Comment:
was this intended? if there are multiple errors then it will repeatedly print the error message